experimental/generation/generator/src/library/integration.ts

```diff
@@ -0,0 +1,734 @@
+/*!
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import * as fs from 'fs-extra';
+import * as ppath from 'path';
+import * as os from 'os';
+
+import * as gen from './dialogGenerator'
+
+const { Templates, SwitchCaseBodyContext } = require('botbuilder-lg');
+const LUParser = require('@microsoft/bf-lu/lib/parser/lufile/luParser');
+const sectionOperator = require('@microsoft/bf-lu/lib/parser/lufile/sectionOperator');
+const lusectiontypes = require('@microsoft/bf-lu/lib/parser/utils/enums/lusectiontypes')
+
+export enum FeedbackType {
+    message,
+    info,
+    warning,
+    error
+}
+
+export type Feedback = (type: FeedbackType, message: string) => void
+
+const GeneratorPattern = /\r?\n> Generator: (.*)/m
+const commentHash = ['.lg', '.lu', '.qna']
+const jsonHash = ['.dialog']
+
+/**
+ * @description：Detect if two files have the same content by hashcode
+ * @param oldPath 
+ * @param newPath
+ * @param fileName
+ */
+function isUnchanged(oldPath: string, newPath: string, fileName: string): boolean {
+    let result = false
+    let ext = ppath.extname(fileName)
+    let oldFile = fs.readFileSync(ppath.join(oldPath, fileName), 'utf8')
+    let newFile = fs.readFileSync(ppath.join(newPath, fileName), 'utf8')
+
+    if (commentHash.includes(ext)) {
+        let matchOld = oldFile.match(GeneratorPattern)
+        let matchNew = newFile.match(GeneratorPattern)
+        if (matchOld && matchNew) {
+            let oldHash = matchOld[1]
+            let newHash = matchNew[1]
+            result = oldHash === newHash
+        }
+    } else if (jsonHash.includes(ext)) {
+        let jsonOld = JSON.parse(oldFile)
+        let oldHash = jsonOld.$Generator
+        let jsonNew = JSON.parse(newFile)
+        let newHash = jsonNew.$Generator
+        if (oldHash && newHash) {
+            result = oldHash === newHash
+        }
+    }
+    return result
+}
+
```

Comment:
Should use the one from dialogGenerator.  You already import as gen.

Reply:
This isUnchanged function is kind of different from that of dialogGenerator.